# Trabajo final
### Fuente de análisis: *Life expectancy at birth (indicator) de OECD(2021)*
Realizada a 44 países de todo el mundo en los años del 2016 al 2019

Base de datos: https://data.oecd.org/healthstat/life-expectancy-at-birth.htm

Este trabajo final consistió en tomar la base de estudio y realizar tres gráficos con sus respectivas conclusiones.

**Participantes**
* Alicia Chaquila Cubillas (20183035)
* Ariza Saito (20185490)

In [1]:
import pandas as pd
import altair as alt

In [2]:
esperanza_vida = pd.read_csv("DP_LIVE_05072021045449693.csv")

In [3]:
esperanza_vida.columns = esperanza_vida.columns.str.lower()
esperanza_vida

location indicator subject measure frequency  time  value flag codes
0        AUS   LIFEEXP     MEN      YR         A  2016   80.4        NaN
1        AUS   LIFEEXP     MEN      YR         A  2017   80.5        NaN
2        AUS   LIFEEXP     MEN      YR         A  2018   80.7        NaN
3        AUS   LIFEEXP     MEN      YR         A  2019   80.9        NaN
4        AUS   LIFEEXP     TOT      YR         A  2016   82.5        NaN
..       ...       ...     ...     ...       ...   ...    ...        ...
598      CRI   LIFEEXP   WOMEN      YR         A  2016   82.6        NaN
599      CRI   LIFEEXP   WOMEN      YR         A  2017   82.7        NaN
600      CRI   LIFEEXP   WOMEN      YR         A  2018   82.9        NaN
601      CRI   LIFEEXP   WOMEN      YR         A  2019   83.0          P
602      CRI   LIFEEXP   WOMEN      YR         A  2020   83.2          P

[603 rows x 8 columns]

In [4]:
# limpieza de columnas
esperanza_vida.drop(columns = 'flag codes', inplace = True)
esperanza_vida

location indicator subject measure frequency  time  value
0        AUS   LIFEEXP     MEN      YR         A  2016   80.4
1        AUS   LIFEEXP     MEN      YR         A  2017   80.5
2        AUS   LIFEEXP     MEN      YR         A  2018   80.7
3        AUS   LIFEEXP     MEN      YR         A  2019   80.9
4        AUS   LIFEEXP     TOT      YR         A  2016   82.5
..       ...       ...     ...     ...       ...   ...    ...
598      CRI   LIFEEXP   WOMEN      YR         A  2016   82.6
599      CRI   LIFEEXP   WOMEN      YR         A  2017   82.7
600      CRI   LIFEEXP   WOMEN      YR         A  2018   82.9
601      CRI   LIFEEXP   WOMEN      YR         A  2019   83.0
602      CRI   LIFEEXP   WOMEN      YR         A  2020   83.2

[603 rows x 7 columns]

In [5]:
# se busca saber los valores máximos y mínimos del value
esperanza_vida.sort_values("value", ascending = False, inplace = True)
esperanza_vida

location indicator subject measure frequency  time  value
209      JPN   LIFEEXP   WOMEN      YR         A  2019   87.5
207      JPN   LIFEEXP   WOMEN      YR         A  2017   87.3
208      JPN   LIFEEXP   WOMEN      YR         A  2018   87.3
206      JPN   LIFEEXP   WOMEN      YR         A  2016   87.1
349      ESP   LIFEEXP   WOMEN      YR         A  2019   86.7
..       ...       ...     ...     ...       ...   ...    ...
535      ZAF   LIFEEXP     TOT      YR         A  2016   63.2
534      ZAF   LIFEEXP     MEN      YR         A  2019   60.7
533      ZAF   LIFEEXP     MEN      YR         A  2018   60.5
532      ZAF   LIFEEXP     MEN      YR         A  2017   60.2
531      ZAF   LIFEEXP     MEN      YR         A  2016   59.8

[603 rows x 7 columns]

### Primer gráfico:

In [7]:
# primer gráfico: esperanza de vida de mujeres del año 2019
cond_1 = esperanza_vida['subject'] == 'WOMEN'
cond_2 = esperanza_vida['time'] == 2019

filtro_1 = esperanza_vida[cond_1 & cond_2]

In [8]:
filtro_1

location indicator subject measure frequency  time  value
209      JPN   LIFEEXP   WOMEN      YR         A  2019   87.5
349      ESP   LIFEEXP   WOMEN      YR         A  2019   86.7
221      KOR   LIFEEXP   WOMEN      YR         A  2019   86.3
112      FRA   LIFEEXP   WOMEN      YR         A  2019   85.9
379      CHE   LIFEEXP   WOMEN      YR         A  2019   85.8
196      ITA   LIFEEXP   WOMEN      YR         A  2019   85.7
235      LUX   LIFEEXP   WOMEN      YR         A  2019   85.2
11       AUS   LIFEEXP   WOMEN      YR         A  2019   85.0
97       FIN   LIFEEXP   WOMEN      YR         A  2019   84.8
503      ISR   LIFEEXP   WOMEN      YR         A  2019   84.8
319      PRT   LIFEEXP   WOMEN      YR         A  2019   84.8
364      SWE   LIFEEXP   WOMEN      YR         A  2019   84.8
289      NOR   LIFEEXP   WOMEN      YR         A  2019   84.7
169      ISL   LIFEEXP   WOMEN      YR         A  2019   84.7
182      IRL   LIFEEXP   WOMEN      YR         A  2019   84.7
529      SVN   LIFEEXP   WOMEN      YR         A  2019   84.5
40       BEL   LIFEEXP   WOMEN      YR         A  2019   84.3
139      GRC   LIFEEXP   WOMEN      YR         A  2019   84.2
53       CAN   LIFEEXP   WOMEN      YR         A  2019   84.2
25       AUT   LIFEEXP   WOMEN      YR         A  2019   84.2
275      NZL   LIFEEXP   WOMEN      YR         A  2019   83.9
125      DEU   LIFEEXP   WOMEN      YR         A  2019   83.7
262      NLD   LIFEEXP   WOMEN      YR         A  2019   83.7
82       DNK   LIFEEXP   WOMEN      YR         A  2019   83.5
440      CHL   LIFEEXP   WOMEN      YR         A  2019   83.4
404      GBR   LIFEEXP   WOMEN      YR         A  2019   83.1
466      EST   LIFEEXP   WOMEN      YR         A  2019   83.0
601      CRI   LIFEEXP   WOMEN      YR         A  2019   83.0
67       CZE   LIFEEXP   WOMEN      YR         A  2019   82.2
304      POL   LIFEEXP   WOMEN      YR         A  2019   81.9
416      USA   LIFEEXP   WOMEN      YR         A  2019   81.4
392      TUR   LIFEEXP   WOMEN      YR         A  2019   81.3
334      SVK   LIFEEXP   WOMEN      YR         A  2019   81.2
586      LTU   LIFEEXP   WOMEN      YR         A  2019   81.2
571      LVA   LIFEEXP   WOMEN      YR         A  2019   80.1
556      COL   LIFEEXP   WOMEN      YR         A  2019   79.9
154      HUN   LIFEEXP   WOMEN      YR         A  2019   79.7
428      BRA   LIFEEXP   WOMEN      YR         A  2019   79.6
452      CHN   LIFEEXP   WOMEN      YR         A  2019   79.2
515      RUS   LIFEEXP   WOMEN      YR         A  2019   78.2
248      MEX   LIFEEXP   WOMEN      YR         A  2019   78.0
491      IDN   LIFEEXP   WOMEN      YR         A  2019   74.0
479      IND   LIFEEXP   WOMEN      YR         A  2019   71.0
542      ZAF   LIFEEXP   WOMEN      YR         A  2019   67.7

## Esperanza de vida de las mujeres en el año 2019

**Análisis:**

Según, la Organización Mundial de la Salud, las mujeres siempre han vivido mas años que los hombres. Se afirma que en promedio las mujeres sobrepasan al grupo masculino por 1,4 años. Existen diversos factores que evidencian este hecho, de los cuales uno es la actitud frente a exámenes médicos. Se demuestra que las mujeres son mas dispuestas a realizarse pruebas o seguir terapias. Además, el porcentaje de suicidios, los accidentes automovilísticos y los homicidios son 4 veces más alta que las mujeres. Sin embargo, la expectativa de vida de las mujeres disminuye si carecen de acceso a los servicios de salud y tiene bajos ingresos económicos, lo que sucede en los países de desarrollo (Noticias ONU, 2019).


Bajo este panorama y la base de datos seleccionados de la OECD, se infiere que en el 2019 Japón, un país desarrollado, tuvo un índice de 87 con respecto a la esperanza de vida de las mujeres. Sin embargo, lo mismo no sucedió con Sudáfrica, ya que se le asignó un índice de 67.7 el cuál lo lleva a representar el país con menor índice.

In [10]:
graf1 = alt.Chart(filtro_1,width = 900, height = 500).mark_point().encode(
    alt.X("location:O", title = "Ubicación"),
    alt.Y("value:Q", title = "Valor"),
    color = alt.Color("value",scale = alt.Scale(scheme = 'turbo')),
    tooltip=["location","value"]
).interactive().properties(
    title = {
    "text" : ["Esperanza de vida de mujeres en el año 2019"],
        "subtitle" : ["Fuente: OECD (2021), Life expectancy at birth (indicator)"],
        "color" : "Brown",
        "subtitleColor" : "Black"
    })

graf1.configure_title(
    fontSize = 20,
    font = 'Courier',
    anchor = 'start',
    )

alt.Chart(...)

### Segundo gráfico:

In [12]:
# segundo gráfico: esperanza de vida de hombres y mujeres en Japón
cond_4 = esperanza_vida['location'] == 'JPN'
cond_5 = esperanza_vida['subject'] == 'TOT'
filtro_3 = esperanza_vida[cond_4 & cond_5]

In [13]:
filtro_3

location indicator subject measure frequency  time  value
205      JPN   LIFEEXP     TOT      YR         A  2019   84.4
204      JPN   LIFEEXP     TOT      YR         A  2018   84.3
203      JPN   LIFEEXP     TOT      YR         A  2017   84.2
202      JPN   LIFEEXP     TOT      YR         A  2016   84.1

## Esperanza de vida al nacer en Japón

**Análisis:**

La tasa de vida en Japón ha subido hasta los 84,1 años, lo que conlleva que este país asiático se encuentre en el puesto 4 de los países más longevos del mundo. Esta cifra se debe al sistema de salud que los japoneses han implementado desde la Segunda Guerra Mundial. Actualmente, ellos van al médico un promedio de 13,4 veces al año, mas del doble que los ciudadanos del Reino Unido. A esto se añade que ponen hincapié a la prevención de factores de riesgos más importantes como enfermedades crónicas, consumo de tabaco, obesidad y la tensión arterial no controlada (El universal, 2020).

Asimismo, según la base de datos de la OECD, se puede deducir que desde el año 2016 el índice de esperanza en Japón ha aumentado constantemente en 0.12% cada año hasta el 2019.

In [14]:
graf2 = alt.Chart(filtro_3,width = 500, height = 500).mark_bar().encode(
    alt.X("time:O", title = "Año"),
    alt.Y("value:Q", title = "Valor"),
    color = alt.Color("value",scale = alt.Scale(scheme = 'set1')),
    tooltip=["location","value"]
).interactive().properties(
    title = {
        "text" : ["Esperanza de vida en Japón"],
        "subtitle" : ["Fuente: OECD (2021), Life expectancy at birth (indicator)"],
        "color" : "Olive",
        "subtitleColor" : "black"
    })

graf2.configure_title(
    fontSize = 20,
    font = 'Courier',
    anchor = 'start',
    )

alt.Chart(...)

### Tercer gráfico:

In [15]:
# tercer grafico : esperanza de vida de los hombres en África

cond_5 = esperanza_vida['subject'] == 'MEN'
cond_6 = esperanza_vida['location'] == 'ZAF'
filtro_4 = esperanza_vida[cond_5 & cond_6]

In [16]:
filtro_4

location indicator subject measure frequency  time  value
534      ZAF   LIFEEXP     MEN      YR         A  2019   60.7
533      ZAF   LIFEEXP     MEN      YR         A  2018   60.5
532      ZAF   LIFEEXP     MEN      YR         A  2017   60.2
531      ZAF   LIFEEXP     MEN      YR         A  2016   59.8

## Esperanza de vida de los hombres en Sudáfrica

**Análisis:**

El acceso a la salud no es prioridad para los residentes donde la esperanza de vida en la más baja. Esta cuestión no solo es discutible en términos económicos o sanitarios, sino que abarca aspectos diversos en la calidad de vida de los africanos. La tasa de mortalidad baja se debe a las enfermedades como infecciones intestinales, respiratorias y malaria. Lo que significa que estas personas no morirían si vivieran en países pobres, territorio y clima que favorecen a distintas plagas y enfermedades. Lamentablemente, vivir en un país con grandes desigualdades no solo en la región, sino en el mismo país, dificulta el avance del sistema sanitario (La vanguardia, 2017).

En este contexto social y de acuerdo a la base de datos seleccionada, la OECD indica que en el 2019 el indice de vida de los hombres en Sudafrica aumentó levemente en 1.38% respecto al 2016, año inicial de estudio.

In [17]:
graf_3 = alt.Chart(filtro_4,width = 500, height = 500).mark_square().encode(
    alt.X("time:O", title = "Año"),
    alt.Y("value:Q", title = "Valor"),
    color = alt.Color("value",scale = alt.Scale(scheme = 'tableau10')),
    tooltip=["location","value"]
).interactive().properties(
    title = {
        "text" : ["Esperanza de vida de los hombres en África"],
        "subtitle" : ["Fuente: OECD (2021), Life expectancy at birth (indicator)"],
        "subtitleColor" : "Black"
    })

graf_3.configure_title(
    fontSize = 20,
    font = 'Courier',
    anchor = 'start',
    color = 'gray')

alt.Chart(...)